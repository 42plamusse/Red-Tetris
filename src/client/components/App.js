import React from "react";
import { hot } from "react-hot-loader";
import { HashRouter as Router, Route } from "react-router-dom";
import EventListener from "react-event-listener";
import { connect } from "react-redux";
import Home from "./Home";
import ErrorPage from "./ErrorPage";
import { setHashError } from "../actions/actions";
import { setConfig } from "react-hot-loader"; //to remove
import Header from "./Header";

setConfig({
  reloadHooks: false
});

let App = ({ hashError, setHashError }) => {
  const handleHashChange = () => {
    const regexp = /^\w{1,12}\[\w{1,12}\]$/;
    const hash = window.location.hash.substr(1);
    const found = hash.match(regexp);
    if (!hash.trim() || found) {
      //no error if no hash or regex match
      setHashError(false);
      return;
    }
    setHashError(true);
  };

  const render = () => {
    if (hashError) {
      return (
        <div>
          <EventListener target="window" onHashChange={handleHashChange} />
          <ErrorPage />;
        </div>
      );
    } else {
      return (
        <div>
          <EventListener target="window" onHashChange={handleHashChange} />
          <Header />
          <Router hashType="noslash">
            <Route path="/" component={Home} />
          </Router>
        </div>
      );
    }
  };

  handleHashChange();
  return render();
};

const mapStateToProps = state => {
  return {
    hashError: state.hashError
  };
};

const actionCreators = { setHashError };

App = connect(
  mapStateToProps,
  actionCreators
)(App);

export default hot(module)(App);
