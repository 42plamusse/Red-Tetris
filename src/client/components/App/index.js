import React from "react";
import { connect } from "react-redux";
import { hot } from "react-hot-loader";
import { setConfig } from "react-hot-loader"; //to remove
import "react-toastify/dist/ReactToastify.css";
import AppSub from "./subcomponent";

setConfig({
  reloadHooks: false
});

const mapStateToProps = state => {
  return {
    playerName: state.player.name,
    roomName: state.room.name
  };
};

export default connect(mapStateToProps)(hot(module)(AppSub));
