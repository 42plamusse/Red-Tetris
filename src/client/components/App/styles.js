import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  display: flex;
  flex-direction: column;
`;
