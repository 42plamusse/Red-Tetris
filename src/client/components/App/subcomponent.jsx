import React from "react";
import Lobby from "../Lobby";
import Header from "../Header";
import Game from "../Game";
import Login from "../Login";
import { StyledContainer } from "./styles";

const AppSub = ({ playerName, roomName }) => {
  return (
    <StyledContainer>
      <Header />
      {!playerName ? (
        // <Popup open={true} hideBackdrop>
        <Login />
      ) : // </Popup>
      !roomName ? (
        <Lobby />
      ) : (
        <Game />
      )}
    </StyledContainer>
  );
};

export default AppSub;
