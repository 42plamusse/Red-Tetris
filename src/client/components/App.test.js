import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { App } from "./App";

describe("<App />", () => {
  test("test App rendering with a player in game", () => {
    const wrapper = shallow(<App playerName="Adubedat" currentRoom="Room1" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("test App rendering with a player in Lobby", () => {
    const wrapper = shallow(<App playerName="Adubedat" currentRoom="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("test App rendering unconnected user", () => {
    const wrapper = shallow(<App playerName="" currentRoom="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
