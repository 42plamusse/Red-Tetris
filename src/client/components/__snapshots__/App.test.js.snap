// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App rendering with a player in Lobby 1`] = `
<div>
  <Connect(HeaderSub) />
  <Connect(LobbySub) />
</div>
`;

exports[`<App /> App rendering with a player in game 1`] = `
<div>
  <Connect(HeaderSub) />
  <Connect(GameSub)
    room="Room1"
  />
</div>
`;

exports[`<App /> App rendering with an unconnected user 1`] = `
<div>
  <Connect(HeaderSub) />
  <Popup
    hideBackdrop={true}
    open={true}
  >
    <Connect(LoginSub) />
  </Popup>
</div>
`;

exports[`<App /> App rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <App
    currentRoom=""
    dispatch={[Function]}
    playerName=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
